android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt

```diff
@@ -49,24 +70,70 @@ class GraphhooperRoutePlugin : MethodCallHandler {
             this.result = result
             this.mapPath = mapPath
         }
 
         private fun ifMapFileExists(): Boolean {
-            return getMapFolder().exists()
+            Log.d("IFMAPFILEEXISTS: ", File(Environment.getExternalStorageDirectory(), "/Download/nepal-gh/").exists().toString())
+            return File(Environment.getExternalStorageDirectory(), "/Download/nepal-gh").exists()
+        }
+
+        private fun downloadMapFromRemoteSource() {
+//            val baseUrl = ""
+//            val fileName = "nepal-gh.zip"
+//            val url = baseUrl
+//
+//            val request = DownloadManager.Request(Uri.parse(url))
+//            request.setAllowedNetworkTypes(DownloadManager.Request.NETWORK_MOBILE or DownloadManager.Request.NETWORK_WIFI)
+//            request.setTitle("Downloading $fileName")
+//            request.setDescription("Please wait")
+//            request.allowScanningByMediaScanner()
+//            request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
+//            request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS,"nepal-gh")
+//
+//            val manager = ctx.applicationContext.getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
+//            downloadId = manager.enqueue(request)
+            unzip(Environment.getExternalStorageDirectory().absolutePath + "/Download/nepal-gh.zip", Environment.getExternalStorageDirectory().absolutePath + "/Download/")
         }
 
-        private fun downloadMapFromRemoteSource(): String {
-            return "Map file is not found"
+        val onDownloadComplete = object : BroadcastReceiver() {
+            override fun onReceive(contxt: Context?, intent: Intent?) {
+                Log.d("download: ", "DOWNLOAD COMPLETED")
+                unzip(Environment.getExternalStorageDirectory().absolutePath + "/Download/nepal-gh.zip", Environment.getExternalStorageDirectory().absolutePath + "/Download/")
+            }
+        }
+
+        private fun unzip(zipFile: String, targetPath: String) {
+            Log.d("UNZIP", targetPath)
+            Log.d("UNZIP", zipFile)
+
+            val zip = ZipFile(zipFile)
+            val enumeration = zip.entries()
+            while (enumeration.hasMoreElements()) {
+                val entry = enumeration.nextElement()
+                val destFilePath = File(targetPath, entry.name)
+                destFilePath.parentFile.mkdirs()
+                if (entry.isDirectory)
+                    continue
+                val bufferedIs = BufferedInputStream(zip.getInputStream(entry))
+                bufferedIs.use {
+                    destFilePath.outputStream().buffered(1024).use { bos ->
+                        bufferedIs.copyTo(bos)
+                    }
+                }
+            }
+            Log.d("MPOINTS: ", mpoints.toString())
+            CalcPath().doInBackground(mpoints.get(0), mpoints.get(1), mpoints.get(2), mpoints.get(3))
```

Comment:
**Known Issue:**
Tried calling the function once the new map is unzipped in the Download. But returns error saying File not found.
After a hot reload everything seems to work